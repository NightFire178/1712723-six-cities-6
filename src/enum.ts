export enum placeFavoriteButtonEnum {
  card = `card`,
  property = `property`
}

export enum cardPlaceEnum {
  cities = `cities`,
  near = `near`,
  favorites = `favorites`
}
