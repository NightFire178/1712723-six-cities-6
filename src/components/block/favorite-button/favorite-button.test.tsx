import React from 'react'
import {render} from '@testing-library/react'
import FavoriteButton from "./favorite-button";
import mockUser from "../../../mock/mock-user";
import {placeFavoriteButtonEnum} from "../../../enum";

const appStateSelection = {
  isAuth:()=>({
    now:true,
    user:mockUser
  })
}

jest.mock("../../../redux/selectors/app-state", ()=>appStateSelection)

const hotelsSelection = {
  hotelFavorites: ()=>true
}

jest.mock("../../../redux/selectors/hotels", ()=>hotelsSelection)

const thunkButtonFavorites = ()=>true
jest.mock('../../../hooks/use-thunk',() =>()=>({thunkButtonFavorites}))

test(`favoriteButton component test`, () => {
  const mockId = 4
  const card = render(<FavoriteButton cardId={mockId} buttonPlace={placeFavoriteButtonEnum.card}/>)
  expect(card).toMatchSnapshot()
  const property = render(<FavoriteButton cardId={mockId} buttonPlace={placeFavoriteButtonEnum.property}/>)
  expect(property).toMatchSnapshot()
})
