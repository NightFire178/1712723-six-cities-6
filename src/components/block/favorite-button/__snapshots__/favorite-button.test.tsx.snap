// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favoriteButton component test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          class="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlink:href="#icon-bookmark"
          />
        </svg>
        <span
          class="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="place-card__bookmark-button place-card__bookmark-button--active button"
      type="button"
    >
      <svg
        class="place-card__bookmark-icon"
        height="19"
        width="18"
      >
        <use
          xlink:href="#icon-bookmark"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`favoriteButton component test 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          class="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlink:href="#icon-bookmark"
          />
        </svg>
        <span
          class="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div>
      <button
        class="property__bookmark-button--active property__bookmark-button button"
        type="button"
      >
        <svg
          class="property__bookmark-icon"
          height="30"
          width="31"
        >
          <use
            xlink:href="#icon-bookmark"
          />
        </svg>
        <span
          class="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="property__bookmark-button--active property__bookmark-button button"
      type="button"
    >
      <svg
        class="property__bookmark-icon"
        height="30"
        width="31"
      >
        <use
          xlink:href="#icon-bookmark"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
